### PR TITLE
Add location helper functions for summary stub and footer sections

### PR DESCRIPTION
**Pull request by rich-iannone**
This PR introduces new location helper functions for the stub parts of a summary (group-wise summary and grand summary) and for footnotes and source notes.

Fixes: https://github.com/rstudio/gt/issues/473
Fixes: https://github.com/rstudio/gt/issues/602

### CODE REVIEW

File: R/location_methods.R [hunk 1]
```diff
@@ -131,7 +135,8 @@ add_summary_location_row <- function(loc,
 add_grand_summary_location_row <- function(loc,
                                            data,
                                            style,
-                                           df_type = "styles_df") {
+                                           df_type = "styles_df",
+                                           in_stub = FALSE) {
```
Comment (jcheng5) [inline R/location_methods.R:139]: Let's infer `in_stub` from the class of `loc`

File: R/location_methods.R [hunk 2]
```diff
@@ -145,16 +150,19 @@ add_grand_summary_location_row <- function(loc,
...
+        expr = !!loc$columns,
+        data = data
+      )
+
+    if (length(columns) == 0) {
+      stop("The location requested could not be resolved:\n",
+           " * Review the expression provided as `columns`",
+           call. = FALSE)
+    }
   }
```
Comment (jcheng5) [inline R/location_methods.R:166]: Let's have an `else` here that sets `columns <- NA_character_`

File: R/utils_render_footnotes.R [hunk 3]
```diff
@@ -165,7 +165,8 @@ resolve_footnotes_styles <- function(data,
       ) %>%
       dplyr::mutate(rownum = (rownum / 100) + row_end) %>%
       dplyr::select(-row_start, -row_end) %>%
-      dplyr::mutate(colnum = colname_to_colnum(data = data, colname = colname))
+      dplyr::mutate(colnum = colname_to_colnum(data = data, colname = colname)) %>%
```
Comment (jcheng5) [inline R/utils_render_footnotes.R:168]: Let's have `colname_to_colnum` convert NA to 0--if this is only needed sometimes then make it conditional on a required argument

File: R/utils_render_html.R [hunk 4]
```diff
@@ -1023,6 +1025,22 @@ create_source_notes_component_h <- function(data) {
...
+
+    source_notes_styles <-
+      if (nrow(source_notes_style) > 0) {
+        source_notes_style$html_style
+      } else {
+        NULL
+      }
+
+  } else {
+    source_notes_styles <- NA_character_
```
Comment (jcheng5) [inline R/utils_render_html.R:1041]: This should be `NULL`

File: R/utils_render_html.R [hunk 5]
```diff
@@ -1023,6 +1025,22 @@ create_source_notes_component_h <- function(data) {
...
   }
 
+  # Get the style attrs for the source notes
+  if ("source_notes" %in% styles_tbl$locname) {
+
+    source_notes_style <- dplyr::filter(styles_tbl, locname == "source_notes")
+
+    source_notes_styles <-
+      if (nrow(source_notes_style) > 0) {
+        source_notes_style$html_style
```
Comment (jcheng5) [inline R/utils_render_html.R:1035]: This should be `paste(collapse = " ", source_notes_style$html_style)`

File: R/utils_render_html.R [hunk 6]
```diff
@@ -1072,6 +1093,21 @@ create_footnotes_component_h <- function(data) {
     dplyr::select(fs_id, footnotes) %>%
     dplyr::distinct()
 
+  # Get the style attrs for the footnotes
+  if ("footnotes" %in% styles_tbl$locname) {
+    footnotes_style <- dplyr::filter(styles_tbl, locname == "footnotes")
+
+    footnotes_styles <-
+      if (nrow(footnotes_style) > 0) {
+        footnotes_style$html_style
```
Comment (jcheng5) [inline R/utils_render_html.R:1102]: Same here--`paste(collapse = " ", footnotes_style$html_style)`

File: R/utils_render_html.R [hunk 7]
```diff
@@ -1072,6 +1093,21 @@ create_footnotes_component_h <- function(data) {
...
+      } else {
+        NULL
+      }
+
+  } else {
+    footnotes_styles <- NA_character_
```
Comment (jcheng5) [inline R/utils_render_html.R:1108]: should be NULL

**Review by jcheng5 — CHANGES_REQUESTED** [review]
See comments.

File: R/utils_render_html.R (hunk 4 above)
Comment (rich-iannone) [reply R/utils_render_html.R:1041]: Fixed.

File: R/utils_render_html.R (hunk 5 above)
Comment (rich-iannone) [reply R/utils_render_html.R:1035]: Done.

File: R/utils_render_html.R (hunk 6 above)
Comment (rich-iannone) [reply R/utils_render_html.R:1102]: Done.

File: R/utils_render_html.R (hunk 7 above)
Comment (rich-iannone) [reply R/utils_render_html.R:1108]: Now changed. Checked the code elsewhere in the file and `NA_character_` was only used in this PR (i.e., `NULL` is used consistently elsewhere).

File: R/location_methods.R (hunk 1 above)
Comment (rich-iannone) [reply R/location_methods.R]: Now done.

File: R/location_methods.R (hunk 2 above)
Comment (rich-iannone) [reply R/location_methods.R:168]: Done.

**Review by jcheng5 — APPROVED** [review]

**Pull request merged**